internal/util/reports_test.go

```diff
@@ -152,14 +227,63 @@ func TestReporterRun(t *testing.T) {
 				"id=6acb7447-eedf-4815-a193-d714c5108f7b;hn=example.local;",
 		}
 		for _, expect := range serializedContent {
-			buffer := make([]byte, 1024)
-			n, _, err := conn.ReadFromUDP(buffer)
-			assert.Equal(len(expect), n)
-			assert.Nil(err)
-			assert.Equal(expect, string(bytes.Trim(buffer, "\x00")))
-
+			received, ok := <-reqs
+			assert.True(ok)
+			short := string(bytes.Trim(received, "\x00"))
+			assert.Equal(len(expect), len(short))
+			assert.Equal(expect, short)
 		}
 		close(done)
 	}()
 	wg.Wait()
 }
+
+// getTLSListener builds a TLS listener using the test certificates
+func getTLSListener() (net.Listener, error) {
+	testCertificate, err := tls.X509KeyPair([]byte(tlsPair.Cert), []byte(tlsPair.Key))
+	if err != nil {
+		return nil, err
+	}
+	conf := &tls.Config{
+		Certificates: []tls.Certificate{
+			testCertificate,
+		},
+		MinVersion: tls.VersionTLS12,
+	}
+	listen, err := tls.Listen("tcp", net.JoinHostPort(reportsHost, strconv.FormatUint(uint64(reportsPort), 10)), conf)
+	if err != nil {
+		return nil, err
+	}
+	return listen, nil
+}
```

Comment:
no-blocker: no need to take action _right now_, but maybe a follow up. Helper functions like this might be good/useful in KTF (e.g. new package `pkg/utils/networking/tls`).